### DISCUSSION TITLE
Build a form/plugin to detect the differences between two points (new - last)

### DISCUSSION THREAD

**Discussion by Bernardodufutur65** [opengisch/QField#5901] [Q&A]
### My project is to show the differences between an old paint from a layer and a new point, add with a GNSS Emlid RS2 on the field with Qfield.
I don't know how to do. Can I do everything if I create a plugin in Qfield?
Or can I do it with a form?

### Also can we add the coordinates directly take by the GNSS in the attribute table?

I do many research projects, but I don't find any answer or kind of same project like me. 
If anyone knows anything related by my questions, please help me. 

Thanks you   (I'm French, sorry for my English )

**Comment by VxTedxV** [comment]
Do you need to look at the result on the field ? Otherwise, QGis can do it after you collect the 2 (or more) points from QField. It's a processing to calculate the shortest way between entities.
Avez-vous besoin de voir le résultat sur le terrain ? Sinon QGis peut faire ce calcul après avoir collecté les 2 points (ou plus) depuis QField. C'est un géotraitement pour calculer la plus petite distance entre deux entités.

**Reply by Bernardodufutur65** [reply]
Thanks for your reply and for the traduction, I appreciate. 
Yes, I need to see the result on the field. It doesn't matter if it's at the end of the survey (after taking all the points) or for each point (point by point).

**Reply by VxTedxV** [reply]
Are old/new points on different layers ? Are the old/new points collected the same day ? Are the old/new point far from each other ? Could there be multiple points close altogether ? In order to help you, all theses questions can be useful.
Est-ce que les anciens/nouveaux points sont sur différentes couches ? Est-ce qu'un ancien point et un nouveau point sont collectés le même jour ? Est-ce que l'ancien point et le nouveau point sont éloignés l'un de l'autre ? Est-ce qu'il peut y avoir plusieurs points proches les uns des autres ? Afin de vous aider dans la démarche, toutes ces questions peuvent être utiles.

**Reply by Bernardodufutur65** [reply]
For now every point is on the same layer but, old and new point are not collected the same day. The distance between them is aproximatively 1cm to 1m.  In the future there will be multiple points close together. Feel good to ask any other question if you want to understand more my subject.

Pour le moment tous les points sont dans la même couche mais les anciens et les nouveau points ne sont pas relevé le même jour. Leurs distance entre eux est entre 1cm et 1m. Plus tard il y aura bien plusieurs points qui seront proche entre eux. N'hésitez pas si vous avez d'autre question, pour comprendre un peu mieux mon sujet.

**Comment by paul-carteron** [comment]
Does the data need to be persistent? If it's for information only, you can use the [measuring tool](https://docs.qfield.org/how-to/measuring-tool/).

If you need it to be persistent, you can ask user to store the value in a layer with three fields (first_point, second_point, distance).

![image](https://github.com/user-attachments/assets/fd181a0a-141c-418f-8e65-d70fe07e0862)

**Reply by Bernardodufutur65** [reply]
sorry for the late. It should be variable data but i can't use the measuring tool because i need to do that automaticaly on all my database in real time. (can't do it manualy) 
For now i have store the information in a persistent field "dist". I do a form on Qgis that calculate the distance between those two point base one the nearest neighbor.

**Comment by ThomasDV1984** [comment]
Cant you use any of the overlay functions in the qgis expression builder?

**Reply by Bernardodufutur65** [reply]
yes i finaly use an expression for that and it work pretty well.
